diff --git a/target/linux/ramips/image/mt7621.mk b/target/linux/ramips/image/mt7621.mk
index 1a8df617b5..5a4a47e011 100644
--- a/target/linux/ramips/image/mt7621.mk
+++ b/target/linux/ramips/image/mt7621.mk
@@ -198,6 +198,20 @@ define Device/alfa-network_ax1800rm
 endef
 TARGET_DEVICES += alfa-network_ax1800rm
 
+define Device/zte_e8820v2
+  $(Device/dsa-migration)
+  MTK_SOC := mt7621
+  DTS := mt7621_zte_e8820v2
+  DEVICE_TITLE := ZTE_E8820v2  
+  IMAGE_SIZE := 16064k
+  DEVICE_VENDOR := ZTE
+  DEVICE_MODEL := E8820V2
+  DEVICE_VARIANT := 16M
+  DEVICE_PACKAGES := kmod-mt7603 kmod-mt76x2 kmod-usb3 kmod-usb-ledtrig-usbport wpad hostapd-utils luci
+endef
+TARGET_DEVICES += zte_e8820v2
+
+
 define Device/alfa-network_quad-e4g
   $(Device/dsa-migration)
   IMAGE_SIZE := 16064k
diff --git a/target/linux/ramips/mt7621/base-files/etc/board.d/01_leds b/target/linux/ramips/mt7621/base-files/etc/board.d/01_leds
index ae46ddb79f..ae39b22c88 100644
--- a/target/linux/ramips/mt7621/base-files/etc/board.d/01_leds
+++ b/target/linux/ramips/mt7621/base-files/etc/board.d/01_leds
@@ -7,6 +7,14 @@ board=$(board_name)
 board_config_update
 
 case $board in
+zte,e8820v2)
+  ucidef_set_led_netdev "sys" "SYS_LED" "$boardname:white:sys" "eth0" "tx rx"
+  ucidef_set_led_default "power" "POWER_LED" "$boardname:white:power" "1"
+;;
+
+
+
+
 alfa-network,ax1800rm)
 	ucidef_set_led_netdev "lan1" "lan1" "green:lan1" "lan1"
 	ucidef_set_led_netdev "lan2" "lan2" "green:lan2" "lan2"
