--- 02_network	2024-05-29 10:14:19.219523000 +0800
+++ 02_network.E8822.openwrt	2024-05-29 12:40:07.753803400 +0800
@@ -8,6 +8,10 @@
 	local board="$1"
 
 	case $board in
+	ZTE,E8822)
+		ucidef_add_switch "switch0" \
+			"0:lan:1" "1:lan:2" "2:lan:3" "3:lan:4" "4:wan:5" "6@eth0"
+		;;	
 	alfa-network,ax1800rm|\
 	ampedwireless,ally-r1900k|\
 	asus,rt-ax53u|\
@@ -181,6 +185,9 @@
 	local label_mac=""
 
 	case $board in
+	ZTE,E8822)
+		wan_mac=$(mtd_get_mac_binary factory 0xe006)
+		;;
 	alfa-network,ax1800rm|\
 	jcg,y2|\
 	wavlink,wl-wn531a6|\
