--- mt7621.mk	2024-05-29 10:14:19.219523000 +0800
+++ mt7621.mk.E8822.openwrt	2024-05-29 12:15:10.065522500 +0800
@@ -198,6 +198,15 @@
   IMAGE/sysupgrade.bin := sysupgrade-tar | append-metadata
 endef
 
+define Device/ZTE_E8822
+  IMAGE_SIZE := 16064k
+  DEVICE_VENDOR := ZTE
+  DEVICE_MODEL := E8822
+  DEVICE_PACKAGES := kmod-mt7603 kmod-mt76x2 kmod-usb2 kmod-usb3 \
+	kmod-usb-ledtrig-usbport wpad-openssl
+endef
+TARGET_DEVICES += ZTE_E8822
+
 define Device/adslr_g7
   $(Device/dsa-migration)
   IMAGE_SIZE := 16064k
