--- ./mt7621.mk	2024-05-29 04:58:29.308307000 +0800
+++ ./mt7621.mk.E8822	2024-05-29 05:53:01.649028600 +0800
@@ -199,6 +199,15 @@
 endef
 TARGET_DEVICES += adslr_g7
 
+define Device/ZTE_E8822
+  IMAGE_SIZE := 16064k
+  DEVICE_VENDOR := ZTE
+  DEVICE_MODEL := E8822
+  DEVICE_PACKAGES := kmod-mt7603 kmod-mt76x2 kmod-usb2 kmod-usb3 \
+	kmod-usb-ledtrig-usbport wpad-openssl
+endef
+TARGET_DEVICES += ZTE_E8822
+
 define Device/afoundry_ew1200
   $(Device/dsa-migration)
   IMAGE_SIZE := 16064k
